lib/rules/no-jquery.js

```diff
@@ -27,60 +31,122 @@ module.exports = {
+        if (isCallExpression(node.parent)) {
+          report(node.parent);
+        } else {
+          report(node);
         }
       },
 
-      VariableDeclarator(node) {
-        if (emberImportAliasName) {
-          destructuredAssignment =
-            node.init && types.isMemberExpression(node.init)
-              ? node.id.name
-              : // assignment of type const $ = Ember.$;
-                utils
-                  .collectObjectPatternBindings(node, {
-                    [emberImportAliasName]: ['$'],
-                  })
-                  .pop();
-        }
-      },
+      /**
+       * Report references of jQuery
+       */
+      Program() {
+        const scope = context.getScope();
+        const tracker = new ReferenceTracker(scope);
 
-      MemberExpression(node) {
-        if (
-          (node.object.name === 'Ember' ||
-            node.object.name === 'Em' ||
-            (emberImportAliasName && node.object.name === emberImportAliasName)) &&
-          node.property.name === '$'
-        ) {
-          report(node);
-        }
+        const globalMap = {
+          $: {
+            [ReferenceTracker.READ]: true,
+          },
+          jQuery: {
+            [ReferenceTracker.READ]: true,
+          },
+          Ember: {
+            $: {
+              [ReferenceTracker.READ]: true,
+            },
+          },
+        };
 
-        if (jqueryImportAliasName && node.object.name === jqueryImportAliasName) {
-          report(node);
-        }
-      },
+        const esmMap = {
+          jquery: {
+            [ReferenceTracker.ESM]: true,
+            default: {
+              [ReferenceTracker.READ]: true,
+            },
+          },
+          ember: {
+            [ReferenceTracker.ESM]: true,
+            default: {
+              $: {
+                [ReferenceTracker.READ]: true,
+              },
+            },
+            $: {
+              [ReferenceTracker.READ]: true,
+            },
+          },
+        };
 
-      CallExpression(node) {
-        if (utils.isGlobalCallExpression(node, destructuredAssignment, ALIASES)) {
-          report(node);
+        /**
+         * Global references
+         * eg; $(body) and $.post()
+         */
+        for (const { node } of tracker.iterateGlobalReferences(globalMap)) {
+          let scopeVariable;
+
+          if (isVariableDeclarator(node.parent)) {
+            // const jq = Ember.$
+            scopeVariable = findVariable(scope, node.parent.id);
+          } else if (isObjectPattern(node.parent)) {
+            // const { $: jq } = Ember;
+            scopeVariable = findVariable(scope, node.value);
+          }
+
+          if (scopeVariable) {
+            // Ignore Variable Declaration
+            const [, ...references] = scopeVariable.references;
+            references.forEach((reference) => {
+              report(reference.identifier.parent);
+            });
```

Comment:
This will ignore the..

```js
cont $ = Ember.$
```

..lines, but will report any and all references to `$` after that